### ISSUE TITLE
Manage home: Request table screen reader output

### ISSUE THREAD

**Issue by AnnaGingle** [cisagov/getgov#2556]
### Issue description

The homepage's domain request table has a column for the delete button. Depending on the state of the domain request the button may or may not be there. The screen reader output is 
`Delete Action | amtrak.gov |`
While it should be 
`Delete Action | amtrak.gov | Domain request cannot be deleted now. Edit the request for more information.`

![Image](https://github.com/user-attachments/assets/42b17cc0-e76a-4451-ac46-7a0a2435ec56)

![Image](https://github.com/user-attachments/assets/035e0e0d-0da5-458f-ba76-ad1b0b9c5ae5)

### Acceptance criteria
- [ ] The screen reader output should be 
`Delete Action | amtrak.gov | Domain request cannot be deleted now. Edit the request for more information.`

### Prioritization Comments
**WCAG Conformance Level:** 2.1 AA
**WCAG Standard that applies:** 
4.1.2 Name, Role, Value (Critical)
**Critical Failure if Not Met?:**  Yes
**Prioritization:** Blocker
**Reason:** This issue hinders the usability of the domain request table for users who rely on screen readers. The current output does not provide sufficient context regarding the action that can or cannot be taken, which can lead to confusion and mistakes.

**Comment by PaulKuykendall** [comment]
According to our state diagram, a domain request can only be deleted from the started, or withdrawn, state. So, at the moment, it is by design that the user does not have the option to delete the other domain request.

https://miro.com/app/board/uXjVMuqbLOk=/?moveToWidget=3458764594819017396&cot=14

**Comment by AnnaGingle** [comment]
This issue will be completed as a part of issue #2848

**Comment by AnnaGingle** [comment]
@PaulKuykendall Not visually, but we need to tell screen reader users the reason there isn't an action available.
